client/src/Components/Widget0.js

```diff
@@ -61,30 +66,69 @@ class Widget0 extends React.Component {
          setInterval(function(){
              fetch('/api/trackleads')
              .then(res => res.json())
-             .then(data => console.log(data))
+             .then(function(data){
+
+                 let {mapStyle} = this.state;
```

Comment:
Can you access `this` in a callback function as a class variable?  That's probably where your error is coming from.

Quick hack would be to create a local reference variable to `this` inside the callback function.